app/lib/common/widgets/emoji_picker_widget.dart

```diff
@@ -83,4 +102,140 @@ class EmojiPickerWidget extends StatelessWidget {
+
+  @override
+  _CustomSearchViewState createState() => _CustomSearchViewState();
+}
+
+/// Default Search View State
+class _CustomSearchViewState extends SearchViewState<_CustomSearchView> {
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(builder: (context, constraints) {
+      final emojiSize =
+          widget.config.emojiViewConfig.getEmojiSize(constraints.maxWidth);
+      final emojiBoxSize =
+          widget.config.emojiViewConfig.getEmojiBoxSize(constraints.maxWidth);
+
+      return Container(
+        color: widget.config.searchViewConfig.backgroundColor,
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Material(
+              color: Colors.transparent,
+              child: SizedBox(
+                height: emojiBoxSize + 8.0,
+                child: Row(
+                  children: [
+                    Expanded(
+                      child: ListView.builder(
+                        padding: const EdgeInsets.symmetric(vertical: 4.0),
+                        scrollDirection: Axis.horizontal,
+                        itemCount: results.length,
+                        itemBuilder: (context, index) {
+                          return buildEmoji(
+                            results[index],
+                            emojiSize,
+                            emojiBoxSize,
+                          );
+                        },
+                      ),
+                    ),
+                    IconButton(
+                      onPressed: () {
+                        widget.closePicker();
+                      },
+                      color: widget.config.searchViewConfig.buttonIconColor,
+                      icon: Icon(PhosphorIcons.xCircle()),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+            Row(
+              children: [
+                IconButton(
+                  onPressed: () {
+                    widget.showEmojiView();
+                  },
+                  color: widget.config.searchViewConfig.buttonIconColor,
+                  icon: const Icon(
+                    Icons.arrow_back,
+                  ),
+                ),
+                Expanded(
+                  child: TextField(
+                    onChanged: onTextInputChanged,
+                    focusNode: focusNode,
+                    style: widget.config.searchViewConfig.inputTextStyle,
+                    decoration: InputDecoration(
+                      border: InputBorder.none,
+                      hintText: widget.config.searchViewConfig.hintText,
+                      hintStyle: widget.config.searchViewConfig.hintTextStyle,
+                      contentPadding:
+                          const EdgeInsets.symmetric(horizontal: 16),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ],
+        ),
+      );
+    },);
```

Comment:
Not enough nesting to break it down in smaller functions?

Reply:
I kept it close to the original source ... I can split it into two sub-widget though.